custom_model_runner/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py

```diff
@@ -53,11 +58,43 @@ def configure(self, params):
 
     def _materialize(self, parent_data_objs, user_data):
         input_filename = self._params["input_filename"]
-        predictions = self._predictor.predict(input_filename)
         output_filename = self._params.get("output_filename")
+
+        additional_params = self._params.get("query_params")
+
         if self._target_type == TargetType.UNSTRUCTURED:
-            with open(output_filename, "w") as f:
-                f.write(predictions)
+            kwargs_params = {}
+
+            # trick in order not to write own parser for params using
+            # "name1=value1;name2=value2" schema
+            _, params = werkzeug.http.parse_options_header("_;" + additional_params)
+
+            with open(input_filename, "rb") as f:
+                data_binary = f.read()
+
+            data_binary_or_text, mimetype, charset = _resolve_incoming_unstructured_data(
+                data_binary,
+                params.get(UnstructuredDtoKeys.MIMETYPE, None),
+                params.get(UnstructuredDtoKeys.CHARSET, None),
+            )
+
+            kwargs_params[UnstructuredDtoKeys.QUERY] = params
+
+            ret_data, ret_kwargs = self._predictor.predict_unstructured(
+                data_binary_or_text, **kwargs_params
+            )
+
+            # only for screen printout convenience we take pred data directly from unstructured_response
+            if isinstance(ret_data, bytes):
+                with open(output_filename, "wb") as f:
+                    f.write(ret_data)
+            else:
+                if ret_data is None:
+                    ret_data = "Return value from prediction is: None (NULL in R)"
+                with open(output_filename, "w") as f:
```

Comment:
be aware that default encoding here is platform dependant. utf-8 on linux

Reply:
FIxed: save file with user's default encoding